app/code/Magento/Catalog/Block/Product/View/Gallery.php

```diff
@@ -170,10 +171,110 @@ public function getGalleryImagesJson()
+            $optionItems['height'] = (int)$this->escapeHtml(
+                $this->getImageAttribute('product_page_image_medium', 'height') ?:
+                $this->getImageAttribute('product_page_image_medium', 'width')
+            );
+        }
+
+        if ($this->getVar("gallery/transition/duration")) {
+            $optionItems['transitionduration'] = (int)$this->escapeHtml($this->getVar("gallery/transition/duration"));
+        }
+
+        $optionItems['transition'] = $this->escapeHtml($this->getVar("gallery/transition/effect"));
+        $optionItems['navarrows'] = $this->getVar("gallery/navarrows");
+        $optionItems['navtype'] = $this->escapeHtml($this->getVar("gallery/navtype"));
+        $optionItems['navdir'] = $this->escapeHtml($this->getVar("gallery/navdir"));
+
+        if ($this->getVar("gallery/thumbmargin")) {
+            $optionItems['thumbmargin'] = (int)$this->escapeHtml($this->getVar("gallery/thumbmargin"));
+        }
+
+        return json_encode($optionItems);
+    }
+
+    /**
+     * Retrieve gallery fullscreen options in JSON format
+     *
+     * @return string
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    public function getGalleryFSOptionsJson()
+    {
+        $fsOptionItems = null;
+
+        //Special case for gallery/nav which can be the string "thumbs/false/dots"
+        is_bool($this->getVar("gallery/fullscreen/nav")) ?
+            $fsOptionItems['nav'] = $this->getVar("gallery/fullscreen/nav") ? 'true' : 'false' :
+            $fsOptionItems['nav'] = $this->escapeHtml($this->getVar("gallery/fullscreen/nav"));
+
+        $fsOptionItems['loop'] = $this->getVar("gallery/fullscreen/loop");
+        $fsOptionItems['navdir'] = $this->escapeHtml($this->getVar("gallery/fullscreen/navdir"));
+        $fsOptionItems['navarrows'] = $this->getVar("gallery/fullscreen/navarrows");
+        $fsOptionItems['navtype'] = $this->escapeHtml($this->getVar("gallery/fullscreen/navtype"));
+        $fsOptionItems['arrows'] = $this->getVar("gallery/fullscreen/arrows");
+        $fsOptionItems['showCaption'] = $this->getVar("gallery/fullscreen/caption");
+
+        if ($this->getVar("gallery/fullscreen/transition/duration")) {
+            $fsOptionItems['transitionduration'] = (int)$this->escapeHtml(
+                $this->getVar("gallery/fullscreen/transition/duration")
+            );
+        }
+
+        $fsOptionItems['transition'] = $this->escapeHtml($this->getVar("gallery/fullscreen/transition/effect"));
+
+        if ($this->getVar("gallery/fullscreen/keyboard")) {
+            $fsOptionItems['keyboard'] = $this->getVar("gallery/fullscreen/keyboard");
+        }
+
+        if ($this->getVar("gallery/fullscreen/thumbmargin")) {
+            $fsOptionItems['thumbmargin'] = (int)$this->escapeHtml($this->getVar("gallery/fullscreen/thumbmargin"));
+        }
+
+        return json_encode($fsOptionItems);
```

Comment:
Please, use `\Magento\Framework\Serialize\Serializer\Json` instead of direct call of the `json_encode ` function